shopfloor_mobile_base/static/wms/src/main.js

```diff
@@ -211,14 +207,59 @@ new Vue({
                 self.appmenu = result.data;
             });
         },
+        login: function (evt, data) {
+            evt.preventDefault();
+            const self = this;
+            this.trigger("login:before");
+            const auth_handler = this._get_auth_handler();
+            if (!_.isUndefined(auth_handler.on_login)) {
+                auth_handler
+                    .on_login(this, evt, data)
+                    .then(this._on_login)
+                    .catch((error) => {
+                        self.trigger("login:failure", error);
+                    });
+            } else {
+                // TODO: we might want to enforce every authentication handler
+                // to provide the on_login handler to avoid such case
+                this._on_login();
+            }
+        },
+        _on_login: function () {
+            const self = this;
+            this._loadConfig().then(function (result) {
```

Comment:
```suggestion
            return this._loadConfig().then(function (result) {
```

be consistent with auth_handler.on_login